File: dealzpark_app/lib/models/shop.dart
class ShopRegistrationData {
  final String shopName;
  final String nid;
  final String tradeLicense;
  final String? productDetails;
  final String? location;
  final String address;
  final String shopType;

  ShopRegistrationData({
    required this.shopName,
    required this.nid,
    required this.tradeLicense,
    this.productDetails,
    this.location,
    required this.address,
    required this.shopType,
  });

  Map<String, dynamic> toJson() {
    return {
      'shopName': shopName,
      'nid': nid,
      'tradeLicense': tradeLicense,
      'productDetails': productDetails,
      'location': location,
      'address': address,
      'shopType': shopType,
    };
  }
}

// If you need to display shop details elsewhere
class Shop {
  final int id;
  final String shopName;
  // ... other fields if needed

  Shop({required this.id, required this.shopName});

  factory Shop.fromJson(Map<String, dynamic> json) {
    return Shop(
      id: json['id'],
      shopName: json['shopName'],
    );
  }
}

File: DealzParkApi/Models/Shop.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace DealzParkApi.Models
{
    public class Shop
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string ShopName { get; set; }

        [Required]
        [MaxLength(50)]
        public string NID { get; set; } // National ID

        [Required]
        [MaxLength(100)]
        public string TradeLicense { get; set; }

        [MaxLength(500)]
        public string ProductDetails { get; set; } // General description of products

        [MaxLength(100)]
        public string Location { get; set; } // e.g., City, Area

        [Required]
        [MaxLength(200)]
        public string Address { get; set; }

        [Required]
        [MaxLength(50)]
        public string ShopType { get; set; } // e.g., Retail, Online, Service

        // Navigation property for EF Core
        [JsonIgnore] // To prevent circular references in serialization
        public virtual ICollection<Offer> Offers { get; set; } = new List<Offer>();
    }
}